test/api/iterations_test.rb

```diff
@@ -132,4 +89,56 @@ def test_skip_problem_as_guest
     assert_equal 401, last_response.status
     assert last_response.body.include?(expected_message)
   end
+
+  private
+
+  def create_submission!(user: nil)
```

Comment:
I'm not a huge fan of anything that makes it easier to create test data. It's very easy to create more than necessary. I'd rather not introduce an external dependency for this.